### ISSUE TITLE
State dict different in train and eval

### ISSUE THREAD

**Issue by vitusbenson** [QUVA-Lab/e2cnn#2]
Just a small problem I just encountered, currently a convolutional models state_dict in train mode is different to its state_dict in eval mode. Which results in a "Missing Keys" error when loading a state_dict saved during training into a model in eval mode. It is not critical though as loading with "strict=False" and then running model.eval() restores the filter and the expanded_bias.
https://github.com/QUVA-Lab/e2cnn/blob/9792ed99582cbd20ba35fee445f11a7c99cc79ab/e2cnn/nn/modules/r2_conv/r2convolution.py#L298

**Comment by Gabri95** [comment]
Hi @vitusbenson,

thanks for reporting this!

This is an expected behaviour because the filter is deleted as soon as model.train() is called (this prevents a user to mistakenly use an older version of the filter and can save some memory during training).

However, I think this kind of behaviour should at least be made more explicit in the documentation.

Thank you for your feedback!

**Comment by Gabri95** [comment]
I've updated the docs (https://quva-lab.github.io/e2cnn/api/e2cnn.nn.html#e2cnn.nn.R2Conv) with a note about this. 
We suggest to always save and load the weights while in the same mode to avoid this issue.

Thanks again for the feedback

Best,
Gabriele
